### PR TITLE
Grammar and Cleanup in Scala 3 Macro Documentation

### PR DESCRIPTION
**Pull request by Zeimyth**
I was trying to familiarize myself with Scala 3 macros today and found multiple issues in the documentation that I felt I could quickly clean up. There are other bigger issues and missing sections, but I tried to limit myself to just cleaning up the grammar of what was already there.

I made a few code snippet changes as well. For these, I usually tried to make things more consistent without trying to reinvent the code in question.

### CODE REVIEW

**Comment by Zeimyth** [comment]
This has merge conflicts now. It's been open for 3 weeks with no activity. If this repo isn't interested in merging small grammar-correcting PRs in a timely manner, I'm of a mind to just close it.

File: _overviews/scala3-macros/best-practices.md [hunk 1]
```diff
@@ -16,24 +16,24 @@ To take the most advantage of the JVM JIT optimisations you want to avoid genera
...
-* Try to avoid `${..}` with arbitrary expressions inside
+* Try to avoid `${...}` with arbitrary expressions inside
   * Use `$someExpr`
   * Use `${ someExprFrom('localExpr) }`
 
 To illustrate, consider the following example:
 ```scala
 val x: StringContext = ...
-'{ StringContext(${Varargs(stringContext.parts.map(Expr(_)))}: _*) }
+'{ StringContext(${Varargs(x.parts.map(Expr(_)))}: _*) }
```
Comment (nicolasstucki) [inline _overviews/scala3-macros/best-practices.md:26]: Maybe we should define the `StringContext` as `sc`

```scala
val sc: StringContext = ...
'{ StringContext(${Varargs(sc.parts.map(Expr(_)))}: _*) }
```

File: _overviews/scala3-macros/tutorial/quotes.md [hunk 2]
```diff
@@ -19,15 +19,14 @@ println(printHello.show) // print("Hello")
 ```
 
 In general, the quote delays the execution while the splice makes it happen before the surrounding code.
-This generalisation allows us to also give meaning to a `${ .. }` that is not within a quote, this evaluate the code within the splice at compile-time and place the result in the generated code.
-Due to some technical considerations we only allow it directly within `inline` definitions that we call a [macro][macros].
+This generalisation allows us to also give meaning to a `${ ... }` that is not within a quote. This evaluates the code within the splice at compile-time and places the result in the generated code.
+Due to some technical considerations, only non-quoted splices are allowed directly within `inline` definitions that we call a [macro][macros].
```
Comment (nicolasstucki) [inline _overviews/scala3-macros/tutorial/quotes.md:23]:
```suggestion
Due to some technical considerations, only top-level splices are allowed directly within `inline` definitions that we call a [macro][macros].
```

File: _overviews/scala3-macros/tutorial/quotes.md [hunk 3]
```diff
@@ -62,58 +61,58 @@ We introduce _levels_ as a count of the number of quotes minus the number of spl
...
-Note that even if we used it implicitly is better to keep it contextual as some changes inside the quote may require it.
+Note that `Type` has a type member called `Underlying` that refers to the type held within the `Type`; in this case, `t.Underlying` is `T`.
+Even if we use the `Type` implicitly, is generally better to keep it contextual as some changes inside the quote may require it.
 The less verbose version is usually the best way to write the types as it is much simpler to read.
-In some cases, we will not know statically the type within the `Type` and will need to use the `.T` to refer to it.
+In some cases, we will not statically know the type within the `Type` and will need to use the `t.Underlying` to refer to it.
 
 When do we need this extra `Type` parameter?
-* When a type is abstract and it is used in a level that is larger than the current level.
+* When a type is abstract and it is used at a level that is deeper than the current level.
```
Comment (nicolasstucki) [inline _overviews/scala3-macros/tutorial/quotes.md:106]:
```suggestion
* When a type is abstract and it is used at a level that is higher than the current level.
```

File: _overviews/scala3-macros/tutorial/quotes.md [hunk 4]
```diff
@@ -267,17 +266,17 @@ def exprOfOption[T: Type](x: Expr[Option[T]])(using Quotes): Option[Expr[T]] =
...
+Now consider the following variant where `x` is an optional value with a (statically) unknown element type:
 
 ```scala
 def exprOfOptionOf[T: Type](x: Expr[Option[Any]])(using Quotes): Option[Expr[T]] =
   x match
     case '{ Some($x: T) } => Some(x) // x: Expr[T]
     case _ => None
 ```
-This time the pattern ` Some($x: T)` will only match if the type of the option is `Some[T]`.
+This time, the pattern `Some($x: T)` will only match if the type of the Option is `Some[T]`.
```
Comment (nicolasstucki) [inline _overviews/scala3-macros/tutorial/quotes.md:279]:
```suggestion
This time, the pattern `Some($x: T)` will only match if the type of the `Option` is `Some[T]`.
```

**Review by nicolasstucki — CHANGES_REQUESTED** [review]
Looks great. Just needs some minor tweaks and a rebase.

**Comment by Zeimyth** [comment]
Thanks for the extra recommended tweaks! Everything has been updated now.

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**